ic-canister/ic-canister-macros/src/api.rs

```diff
@@ -612,7 +611,9 @@ pub(crate) fn generate_idl() -> TokenStream {
                 let mut rets = Vec::new();
                 #(#rets)*
                 let func = Function { args, rets, modes: #modes };
-                service.push((#name.to_string(), Type::Func(func)));
+                if !cfg!(feature = "no_api") || (#name != "burn" && #name != "mint") {
```

Comment:
There should not be check for any specific method names here. This crate is not only used for IS20 tokens. And I don't really get why you want to add these conditions in the first place...

Reply:
If the feature `no_api` is set, all the methods are excluded.
I will make a separate PR for IS20 to propagate the `no_api` feature.

Reply:
@Maximkaaa  This is the IS20 [PR](https://github.com/infinity-swap/IS20/pull/130